### PR TITLE
Sc rn aseq gdc card

### PR DESCRIPTION
**Pull request by congyu-lu**
## Description
need to release to GDC and then create a PR for gff changes once this pr is merged
will add scrna gene expression in another pr

If local GFF is not available: can test by pulling sjpp/master, and opening http://localhost:3000/example.gdc.scRNAseq.html.

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

**Comment by siosonel** [comment]
Screenshot 2024-09-12 at 11 01 27 PM
Screenshot 2024-09-12 at 11 01 27 PM


Screenshot 2024-09-12 at 11 00 30 PM

Screenshot 2024-09-12 at 11 00 30 PM

**Review by siosonel — APPROVED** [review]

File: client/gdc/singlecell.js [hunk 1]
```diff
@@ -111,7 +213,7 @@ async function submitSelectedFile(fileId, obj) {
 
 	let result
 	{
-		const body = { genome: gdcGenome, dslabel: gdcDslabel, sample: fileId }
+		const body = { genome: gdcGenome, dslabel: gdcDslabel, sample: fileId, plots: ['UMAP', 'TSNE', 'PCA'] }
```
Comment (xzhou82) [inline client/gdc/singlecell.js:216]: does this code have access to termdbconfig?

Comment (congyu-lu) [reply client/gdc/singlecell.js:216]: this submitSelectedFile function (and getFilesAndShowTable) are not used. maybe we delete them?

File: client/gdc/singlecell.js [hunk 2]
```diff
@@ -29,6 +33,99 @@ const columns = [
...
+					config: updateArg
+				})
+			}
+		}
+	}
+
+	return api
+}
+
 export async function gdcSinglecellUi({ holder, filter0, callbackOnRender, debugmode = false }) {
```
Comment (xzhou82) [inline client/gdc/singlecell.js:129]: looks like this func is only used in app.parseurl.js. i don't think it's essential to keep this `gdcsinglecell=1` url trigger, so we should delete this function and the trigger so no need to maintain this table rendering code

**Review by xzhou82 — APPROVED** [review]
thanks a lot. please address issues in new branch

File: client/gdc/singlecell.js (hunk 1 above)
Comment (xzhou82) [reply client/gdc/singlecell.js:216]: yes please try. in new branch?

Comment (congyu-lu) [reply client/gdc/singlecell.js:216]: ok sounds good

**Comment by xzhou82** [comment]
please also post the GFF pull request link here thanks

**Comment by congyu-lu** [comment]
> please also post the GFF pull request link here thanks

ok will release to GDC and then make the gff PR (needs to include the latest pp version number)

File: client/plots/singleCellPlot.js [hunk 3]
```diff
@@ -46,16 +46,12 @@ class singleCellPlot {
 		const state = this.getState(appState)
 		const q = state.termdbConfig.queries
 		this.tableOnPlot = appState.nav?.header_mode == 'hidden'
-
-		//read files data
-		const controlsDiv = this.opts.holder
```
Comment (airenzp) [inline client/plots/singleCellPlot.js:51]: Looking at the SingleCellPlot I noticed the controls are not properly placed after this changes. If possible next time add me as a reviewer.

File: client/plots/singleCellPlot.js [hunk 4]
```diff
@@ -212,7 +221,8 @@ class singleCellPlot {
 						}
 						const config = {
 							chartType: 'gsea',
-							gsea_params: gsea_params
+							gsea_params: gsea_params,
+							insertBefore: this.app.opts?.app?.getPlotHolder && this.showDivId
```
Comment (airenzp) [inline client/plots/singleCellPlot.js:225]: Is this ok?. getPlotHolder is a function and insertBefore expects an id

**Pull request merged**